provider/cluster_resource_test.go

```diff
@@ -17,295 +17,155 @@ limitations under the License.
-	})
-
-	It("Sets version", func() {
-		// Prepare the server:
-		server.AppendHandlers(
-			CombineHandlers(
-				VerifyRequest(http.MethodPost, "/api/clusters_mgmt/v1/clusters"),
-				VerifyJQ(".version.id", "openshift-v4.8.1"),
-				RespondWithPatchedJSON(http.StatusOK, template, `[
-				  {
-				    "op": "replace",
-				    "path": "/version",
-				    "value": {
-				      "id": "openshift-v4.8.1"
-				    }
-				  }
-				]`),
-			),
-		)
-
-		// Run the apply command:
-		terraform.Source(`
-		  resource "ocm_cluster" "my_cluster" {
-		    name           = "my-cluster"
-			product		   = "osd"
-		    cloud_provider = "aws"
-		    cloud_region   = "us-west-1"
-		    version        = "openshift-v4.8.1"
-		  }
-		`)
-		Expect(terraform.Apply()).To(BeZero())
-
-		// Check the state:
-		resource := terraform.Resource("ocm_cluster", "my_cluster")
-		Expect(resource).To(MatchJQ(".attributes.version", "openshift-v4.8.1"))
+		clusterState := &ClusterState{
+			Name: types.String{
+				Value: "my-cluster",
+			},
+			CloudRegion: types.String{
+				Value: "us-east-1",
+			},
+			AWSAccountID: types.String{
+				Value: "123456789012",
+			},
+			AvailabilityZones: types.List{
+				Elems: []attr.Value{
+					types.String{
+						Value: "us-east-1a",
+					},
+				},
+			},
+			Properties: types.Map{
+				Elems: map[string]attr.Value{
+					"rosa_creator_arn": types.String{
+						Value: "arn:aws:iam::123456789012:dummy/dummy",
+					},
+				},
+			},
+			Wait: types.Bool{
+				Value: false,
+			},
+		}
+		builder, err := createClusterBuilder(context.Background(), clusterState, diag.Diagnostics{})
+		Expect(err).To(BeNil())
+
+		object, err := builder.Build()
+		Expect(err).To(BeNil())
+		Expect(object).ToNot(BeNil())
+
+		Expect(object.Name()).To(Equal("my-cluster"))
+
+		id, ok := object.Region().GetID()
+		Expect(ok).To(BeTrue())
+		Expect(id).To(Equal("us-east-1"))
+
+		Expect(object.AWS().AccountID()).To(Equal("123456789012"))
+
+		availabilityZones := object.Nodes().AvailabilityZones()
+		Expect(availabilityZones).To(HaveLen(1))
+		Expect(availabilityZones[0]).To(Equal("us-east-1a"))
+
+		arn, ok := object.Properties()["rosa_creator_arn"]
+		Expect(ok).To(BeTrue())
+		Expect(arn).To(Equal("arn:aws:iam::123456789012:dummy/dummy"))
 	})
 
-	It("Fails if the cluster already exists", func() {
-		// Prepare the server:
-		server.AppendHandlers(
-			CombineHandlers(
-				VerifyRequest(http.MethodPost, "/api/clusters_mgmt/v1/clusters"),
-				RespondWithJSON(http.StatusBadRequest, `{
-				  "id": "400",
-				  "code": "CLUSTERS-MGMT-400",
-				  "reason": "Cluster 'my-cluster' already exists"
-				}`),
-			),
-		)
-
-		// Run the apply command:
-		terraform.Source(`
-		  resource "ocm_cluster" "my_cluster" {
-			product		   = "osd"
-		    name           = "my-cluster"
-		    cloud_provider = "aws"
-		    cloud_region   = "us-west-1"
-		  }
-		`)
-		Expect(terraform.Apply()).ToNot(BeZero())
+	It("populates state correctly", func() {
```

Comment:
Add more tests 
Wait & ready
wait & error
wait & not ready

Reply:
We decided to postpone testing it.